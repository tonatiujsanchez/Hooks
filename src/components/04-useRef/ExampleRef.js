import { useState } from "react"
import MultipleCustomHooks from "../03-examples/MultipleCustomHooks"


const ExampleRef = () => {

    const [show, setShow] = useState(false)
    

  return (

    <div>
        <div className="d-flex justify-content-between">
            <h2>Example Ref</h2>
            <button
                    onClick={ ()=> setShow( !show ) } 
                    className="btn btn-secondary">
            Show/Hidden</button>
        </div>
        <hr/>
        { show && <MultipleCustomHooks /> }
        <div className="my-3">

        </div>
    </div>

  )
}

export default ExampleRef