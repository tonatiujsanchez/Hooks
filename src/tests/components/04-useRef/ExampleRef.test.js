import { shallow } from "enzyme"
import ExampleRef from "../../../components/04-useRef/ExampleRef"




describe('Pruebas en <ExampleRef />', () => { 

    test('Debe mostrarse correctamente', () => { 

        const wrapper = shallow( <ExampleRef /> )
        expect( wrapper ).toMatchSnapshot()

     })

     test('Debe de mostrar el componente <MultipleCustomHooks />', () => { 
        const wrapper = shallow( <ExampleRef /> )

        const button = wrapper.find('button')
        button.simulate('click')

        const multipleCustomHooks = wrapper.find('MultipleCustomHooks')
        expect( multipleCustomHooks.exists() ).toBe(true)


      })

 })