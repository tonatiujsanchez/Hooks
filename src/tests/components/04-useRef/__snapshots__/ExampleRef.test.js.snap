// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <ExampleRef /> Debe mostrarse correctamente 1`] = `
<div>
  <div
    className="d-flex justify-content-between"
  >
    <h2>
      Example Ref
    </h2>
    <button
      className="btn btn-secondary"
      onClick={[Function]}
    >
      Show/Hidden
    </button>
  </div>
  <hr />
  <div
    className="my-3"
  />
</div>
`;
